components/dropdown/dropdown-content.js

```diff
@@ -9,6 +9,7 @@ import { dropdownContentStyles } from './dropdown-content-styles.js';
  * @slot footer - Sticky container at the bottom of the dropdown
  * @fires d2l-dropdown-open - Dispatched when the dropdown is opened
  * @fires d2l-dropdown-close - Dispatched when the dropdown is closed
+ * @fires d2l-dropdown-focus-enter - Dispatched when user focus enters the dropdown content
```

Comment:
Can we have some way to specify that this event only fires when the trap-focus option is on? Maybe just a "(trap-focus option only)" at the end of the line. Otherwise consumers might think that it happens whenever the user clicks inside the dropdown.